nilearn/interfaces/fmriprep/tests/test_load_confounds_utils.py

```diff
@@ -25,10 +30,38 @@ def test_sanitize_confounds(inputs, flag):
     assert singleflag is flag
 
 
-@pytest.mark.parametrize("flag,suffix",
-                         [(True, "_desc-confounds_regressors"),
-                          (False, "_desc-confounds_timeseries")])
-def test_get_file_name(tmp_path, flag, suffix):
-    img, _ = create_tmp_filepath(tmp_path, old_derivative_suffix=flag)
+@pytest.mark.parametrize(
+    "flag,suffix,image_type,kwargs",
+    [
+        (True, "_desc-confounds_regressors", "regular", {}),
+        (False, "_desc-confounds_timeseries", "regular", {}),
+        (True, "_desc-confounds_regressors", "native", {}),
+        (False, "_desc-confounds_timeseries", "native", {}),
+        (True, "_desc-confounds_regressors", "res", {}),
+        (False, "_desc-confounds_timeseries", "res", {}),
+        (True, "_desc-confounds_regressors", "cifti", {}),
+        (False, "_desc-confounds_timeseries", "cifti", {}),
+        (True, "_desc-confounds_regressors", "den", {}),
+        (False, "_desc-confounds_timeseries", "den", {}),
+        (
+            True,
+            "_desc-confounds_regressors",
+            "part",
+            {"suffix": "sub-test01_task-test_part-mag_run-01"},
+        ),
+        (
+            False,
+            "_desc-confounds_timeseries",
+            "part",
+            {"suffix": "sub-test01_task-test_part-mag_run-01"},
+        ),
+        (True, "_desc-confounds_regressors", "gifti", {}),
+        (False, "_desc-confounds_timeseries", "gifti", {}),
+    ],
+)
+def test_get_file_name(tmp_path, flag, suffix, image_type, kwargs):
+    img, _ = create_tmp_filepath(
+        tmp_path, image_type=image_type, old_derivative_suffix=flag, **kwargs
+    )
```

Comment:
Since we are almost taking the combination of all the parameters, I would refactor this block to:
```suggestion
@pytest.mark.parametrize("flag", [True, False])
@pytest.mark.parametrize("suffix", ["_desc-confounds_regressors", "_desc-confounds_timeseries"])
@pytest.mark.parametrize("image_type", ["regular", "native", "res", "cifti", "den", "part", "gifti"])
def test_get_file_name(tmp_path, flag, suffix, image_type):
    if image_type == "part":
        kwargs = {"suffix": "sub-test01_task-test_part-mag_run-01"}
    else:
        kwargs = {}

   img, _ = create_tmp_filepath(
        tmp_path, image_type=image_type, old_derivative_suffix=flag, **kwargs
    )

```

Reply:
I love it!

Reply:
Make sure to check the syntax errors that i might have created... it's difficult to get it right in the suggestion mode.